### DISCUSSION TITLE
Hub to PC and the bleak library

### DISCUSSION THREAD

**Discussion by jeagh** [pybricks/support#1040] [Q&A] [answered]
I upgraded my Xubuntu installation to 23.04 today. This version includes Python 11 and now I need to create a virtual environment so that "pip3 install bleak" will do it's job correctly.

After a bit of head scratching I now have bleak installed and if I run the version of python that's in the virtual environment then I can import bleak without any errors.

Now what I need to understand is how to have VScode run my code, which I assume needs to be in the same virtual environment, using the version of python from the same virtual environment. The system wide version of python doesn't, of course, find bleak.

Surly, I don't need to create a virtual environment for each new module that I want to install? I suppose VScode also needs to be in the same virtual environment?

**Comment by dlech** [comment, marked as answer]
If you have the Microsoft Python extension for VS Code installed, you can select the active interpreter using the command palette or the status bar. https://code.visualstudio.com/docs/python/environments#_working-with-python-interpreters

You must install everything you want to use in the same virtual environment.

**Reply by jeagh** [reply]
Thank you David for the link and I am using VS Code. All good now and my pc to hub project is working again.
